snuba/utils/streams/batching.py

```diff
@@ -193,15 +163,12 @@ def _run_once(self) -> None:
         if self.producer:
             self.producer.poll(0.0)
 
+        # TODO: This needs to handle recoverable errors (end of partition,
+        # transport errors, etc.)
```

Comment:
This will probably end up being something like:

```python
try:
    msg = self.consumer.poll(timeout=1.0)
except self.recoverable_errors:
    return None
```

Where "recoverable_errors" is a `Tuple[ConsumerError]` or something, received as a constructor parameter, and passed by the `ConsumerBuilder` (where anything Kafka-specific should live.) The `KafkaConsumer` will throw subclasses of `ConsumerError` for known errors like `EndOfPartitionError(ConsumerError)`, etc. based on the error code received from the `ConfluentMessage`.

Reply:
(This is done now.)